examples/scripts/apply.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash -eu
+#
+# Copyright 2017 The Kapitan Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{% set i = inventory.parameters %}
+DIR=$(dirname ${BASH_SOURCE[0]})
+
```

Comment:
when you create a new namespace for the very first time and pass many resources to `kubectl apply`, it's not smart enough to know it has to create the namespace before everything else.

add this on line 19

```
# Create namespace before anything else
${DIR}/kubectl.sh apply -f ${DIR}/../pre-deploy/01_namespace.yml
```

Reply:
that's why I use the numbers, to force an order when deploying.

Reply:
that was true when we were applying one file at a time. Now that we apply all files in a folder at once, you lose the `01` in the filename and you end up with a huge yml with multiple resources being passed to `apply`.

At least for me, kubectl wasn't smart enough to know it had to apply the namespace before any other resource.

Reply:
It has always worked for me with the number approach.
if we cannot trust it we will have other problems, i.e. patching a service account with a pull secret that has not been created.